src/presenter/trip-list-presenter.js

```diff
@@ -32,50 +31,39 @@ export default class TripListPresenter {
     this.#eventDestinations = [...this.#destinationModel.destination];
     this.#eventOffersByType = [...this.#offerModel.offerByType];
 
-    this.#renderEventPointList();
+    this.#renderTripList();
   };
 
-  #renderEventPoint = (eventPoint, destinations, offersByType) => {
-    const eventPointComponent = new EventPointView(eventPoint, destinations, offersByType);
-    const formEditComponent = new EditFormView(eventPoint, destinations, offersByType);
-
-    const replaceEventPointToEditForm = () => replace(formEditComponent, eventPointComponent);
-
-    const replaceEditFormToEventPoint = () => replace(eventPointComponent, formEditComponent);
-
-    const onEscKeyDown = (evt) => {
-      if (isPressEscape(evt)) {
-        evt.preventDefault();
-        replaceEditFormToEventPoint();
-        document.removeEventListener('keydown', onEscKeyDown);
-      }
-    };
+  #handleModeChange = () => this.#eventPointPresenter.forEach((presenter) => presenter.resetView());
 
-    const onOpenFormEdit = () => {
-      replaceEventPointToEditForm();
-      document.addEventListener('keydown', onEscKeyDown);
-    };
-
-    const onCloseFormEdit = () => {
-      replaceEditFormToEventPoint();
-      document.addEventListener('keydown', onEscKeyDown);
-    };
+  #handleEventPointChange = (updatedPoint) => {
+    this.#eventPointsList = updateItem(this.#eventPointsList, updatedPoint);
+    this.#eventPointPresenter.get(updatedPoint.id).init(updatedPoint, this.#eventDestinations, this.#eventOffersByType);
+  };
 
-    eventPointComponent.setExpandButtonClickHandler(onOpenFormEdit);
+  #renderEventPoint = (eventPoint) => {
+    const eventPointPresenter = new EventPointPresenter(this.#tripListComponent.element, this.#handleEventPointChange, this.#handleModeChange);
+    eventPointPresenter.init(eventPoint, this.#eventDestinations, this.#eventOffersByType);
+    this.#eventPointPresenter.set(eventPoint.id, eventPointPresenter);
+  };
 
-    formEditComponent.setCollapseButtonClickHandler(onCloseFormEdit);
+  #rederEventPoins = () => {
+    this.#eventPointsList.forEach((event) => this.#renderEventPoint(event));
```

Comment:
Можно немного короче
```suggestion
    this.#eventPointsList.forEach(this.#renderEventPoint);
```